packages/devtools-reps/src/reps/function.js

```diff
@@ -29,62 +29,78 @@ function FunctionRep(props) {
   let grip = props.object;
 
   return (
-    span({
+   span({
       "data-link-actor-id": grip.actor,
       className: "objectBox objectBox-function",
       // Set dir="ltr" to prevent function parentheses from
       // appearing in the wrong direction
       dir: "ltr",
     },
-      getTitle(grip, props),
-      getFunctionName(grip, props),
-      "(",
-      ...renderParams(props),
-      ")"
+    getTitle(grip, props),
+    getFunctionName(grip, props),
+    "(",
+    ...renderParams(props),
+    ")"
     )
   );
 }
 
 function getTitle(grip, props) {
-  const {
-    mode
+  const { 
+    mode 
   } = props;
 
   if (mode === MODE.TINY && !grip.isGenerator && !grip.isAsync) {
     return null;
   }
 
-  let title = mode === MODE.TINY
-    ? ""
-    : "function ";
+  let title = mode === MODE.TINY 
+  ? "" 
+  : "function ";
 
   if (grip.isGenerator) {
-    title = mode === MODE.TINY
-      ? "* "
-      : "function* ";
+    title = mode === MODE.TINY 
+    ? "* " 
+    : "function* ";
   }
 
   if (grip.isAsync) {
     title = "async" + " " + title;
   }
 
   return span({
-    className: "objectTitle",
-  }, title);
+      className: "objectTitle",
+    },title);
 }
 
 function getFunctionName(grip, props) {
-  let name = grip.userDisplayName
-    || grip.displayName
-    || grip.name
-    || props.functionName
-    || "";
+  let { functionName } = props;
+  let end = functionName.length - 1;
```

Comment:
this is why you see failure on tests.
Most of the test do not pass the `functionName` props to the component (which is valid).
Because it is undefined, we get the following error : `TypeError: Cannot read property 'length' of undefined`

Here we can make things right like : 
```js
let { functionName } = props;
if (functionName) {
  let end = functionName.length - 1;
  functionName =
    functionName.startsWith('"') && functionName.endsWith('"')
      ? functionName.substring(1, end)
      : functionName;
}

if (
  grip.displayName != undefined 
  && functionName != undefined 
  && grip.displayName != functionName) {
  name = functionName + ": " + grip.displayName;
} else {
  name =
    grip.userDisplayName ||
    grip.displayName ||
    grip.name ||
    props.functionName ||
    "";
}

return cropString(name, 100);
```